HotelBooking.Service/Services/HotelService.cs

```diff
@@ -144,5 +223,84 @@ public async Task<bool> AddServiceAndFacilityToRoomAsync(EquipRoomRequest model)
             await unitOfWork.SaveAsync();
             return true;
         }
+
+        public async Task<IEnumerable<HotelModel>> GetHotelByAddressTypeRoomDuration(FilterHotelRequest model)
+        {
+            var dataSet = hotelRepository.GetAllHotels();
+            var result = await dataSet.ApplyFilterByAddress(model.City)
+                                      .ApplyFilterByRoomType(model.RoomType)
+                                      .Include(x => x.Address)
+                                      .Include(x => x.Urls)
+                                      .Include(x => x.Rooms).ThenInclude(x => x.RoomFacilities).ThenInclude(x => x.Facility)
+                                      .Include(x => x.Rooms).ThenInclude(x => x.RoomServices).ThenInclude(x => x.Service)
+                                      .ToListAsync();
+            var hotels = new List<Hotel>();
+            if (result.Any())
+            {
+
+                foreach (var hotel in result)
+                {
+                    var rooms = await roomRepository.GetByCondition(x => x.HotelId.Equals(hotel.Id) && x.RoomType == model.RoomType).ToListAsync();
+                    if (rooms.Any())
+                    {
+                        foreach (var room in rooms)
+                        {
+                            var checkResult = await checkDurationValidationService.CheckValidationDurationForRoom(model.Duration, room.Id);
+                            if (checkResult == true)
```

Comment:
to much nesty logic. Need to refactor to reduce complexity

Reply:
I fixed it. thanks.